### PR TITLE
Pool party

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Ports the swimming pool from NSV.
Replaces the pool in the luxury shuttle with a real pool.
Replaces the pool in the hotel with a real pool.
Replaces the boxing arena on deltastation with a swimming pool.

Ports
https://github.com/BeeStation/NSV13/pull/781
https://github.com/BeeStation/NSV13/pull/824
https://github.com/BeeStation/NSV13/pull/840
 + Does a bit of mapping
 + Fixes some spelling mistakes and shitcode

Continuing the legacy of the cursed PR, to date every attempt at porting pools has failed.

## Why It's Good For The Game

Swimming is fun. Torturing felinids is fun.

## Changelog
:cl: PowerfulBacon, KMC, Cdey
add: Swimming pools
/:cl:

### CODE REVIEW

**Comment by ToasterBan** [comment]
based

File: code/modules/pool/pool.dm [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+/obj/machinery/pool_filter
+	name = "Pool filter"
+	desc = "A device which can help you regulate conditions in a pool. Use a <b>wrench</b> to change its operating temperature, or hit it with a reagent container to load in new liquid to add to the pool."
+	icon = 'icons/obj/pool.dmi'
+	icon_state = "poolfilter"
+	pixel_y = 12 //So it sits above the water
+	idle_power_usage = IDLE_POWER_USE
+	var/id = null //change this if youre an annoying mapper who wants multiple pools per area.
+	var/list/pool = list()
+	var/desired_temperature = 300 //Room temperature
```
Comment (ghost) [inline code/modules/pool/pool.dm:195]:
```suggestion
	var/desired_temperature = 293 //Room temperature
```
Assuming this is in Kelvin, this is wrong.

File: code/modules/pool/pool.dm [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	var/current_temperature = 300 //current temp
+	var/preset_reagent_type = null //Set this if you want your pump to start filled with a given reagent. SKEWIUM POOL SKEWIUM POOL!
```
Comment (ghost) [inline code/modules/pool/pool.dm:197]:
```suggestion
	var/preset_reagent_type = null //Set this if you want your pump to start filled with a given reagent. 
```

No longer exists in this codebase.

**Comment by Blueturbo47** [comment]
> ## Why It's Good For The Game
> Swimming is fun. Torturing felinids is fun.

Bacon are you... a masochist?

**Comment by ghost** [comment]
Changelog needs the original PR authors.

File: code/modules/pool/pool.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/pool/pool.dm:195]: Room temperature in kelvin ranges from 283 to 303 according to https://www.nist.gov/pml/weights-and-measures/si-units-temperature Should be fine, I think this is only used for checking anyway

**Comment by qwertyquerty** [comment]
asphyxiation fetish detected

**Comment by Penwin0** [comment]
Very good, you can keep your hands

**Comment by theboy6545** [comment]
> Changelog needs the original PR authors.

Bacon made the PR (but he might have had help from others I can't remember)

**Comment by PowerfulBacon** [comment]
> 
> 
> > Changelog needs the original PR authors.
> 
> Bacon made the PR (but he might have had help from others I can't remember)

KMC made the original base code, Cdey made the sprites and I made the swimming components. Added them to the CL

**Comment by ToasterBan** [comment]
haha what if we added this ready

haha jkjk

unless?

File: code/modules/pool/components/swimming.dm [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+			bob_tick = 0
+	if(!floating && is_drowning(L))
+		if(!drowning)
+			start_drowning(L)
+			drowning = TRUE
+		drown(L)
+	else
+		if(drowning)
+			stop_drowning(L)
+			drowning = FALSE
```
Comment (utilbird) [inline code/modules/pool/components/swimming.dm]:
```suggestion
	else if(drowning)
		stop_drowning(L)
		drowning = FALSE
```

File: code/modules/pool/pool.dm [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+		return
+	newTemp = CLAMP(newTemp, T0C, 320)
+	desired_temperature = newTemp
+	return FALSE
+
+/obj/machinery/pool_filter/process()
+	if(!pool?.len)
+		return //No use having one of these processing for no reason is there?
+	if(!is_operational())
+		return
```
Comment (utilbird) [inline code/modules/pool/pool.dm]:
```suggestion
	if(!LAZYLEN(pool) || !is_operational())
		return //No use having one of these processing for no reason is there?
```

File: code/modules/pool/pool_items.dm [hunk 5]
```diff
@@ -0,0 +1,93 @@
+/obj/item/proc/check_float(mob/living/holder)
+	return FALSE
```
Comment (utilbird) [inline code/modules/pool/pool_items.dm]: it seems a bit hacky to have this proc for every single item when only 2 items will ever return true. Wouldn't it be better to have a subtype check (`/obj/item/twohanded/required/pool`) and have the proc for those items only?

File: code/modules/pool/pool.dm [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+			SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "pool", /datum/mood_event/poolparty)
+			return
+		SEND_SIGNAL(user, COMSIG_ADD_MOOD_EVENT, "pool", /datum/mood_event/poolwet)
+
+//Largely a copypaste from shower.dm. Checks if the mob was stupid enough to enter a pool fully clothed. We allow masks as to not discriminate against clown and mime players.
+/turf/open/indestructible/sound/pool/proc/check_clothes(mob/living/carbon/human/H)
+	if(!istype(H)) //Don't care about non humans.
+		return FALSE
+	if(iscatperson(H))
+		return FALSE
```
Comment (utilbird) [inline code/modules/pool/pool.dm]:
```suggestion
	if(!istype(H) || iscatperson(H)) //Don't care about non humans.
		return FALSE
```

File: code/modules/pool/pool.dm [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+		// with a radiation suit on.
+		return FALSE
+
+	. = FALSE
+	if(H.wear_suit && !(H.wear_suit.clothing_flags & SHOWEROKAY))
+		. = TRUE
+	else if(H.w_uniform && !(H.w_uniform.clothing_flags & SHOWEROKAY))
+		. = TRUE
+	else if(H.head && !(H.head.clothing_flags & SHOWEROKAY))
+		. = TRUE
```
Comment (utilbird) [inline code/modules/pool/pool.dm]:
```suggestion
	if(!(H.wear_suit?.clothing_flags & SHOWEROKAY))
		return TRUE
	if(!(H.w_uniform?.clothing_flags & SHOWEROKAY))
		return TRUE
	if(!(H.head?.clothing_flags & SHOWEROKAY))
		return TRUE
```

Comment (utilbird) [reply code/modules/pool/pool.dm]: You could put this all under one check too, if you wanted to. It's a bit long so eh

File: code/modules/pool/pool.dm [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+				else //Sometimes the water penetrates a lil deeper than just a splosh.
+					splash_holder.reaction(M, INGEST)
+				splash_holder.trans_to(M, trans_amount, transfered_by = src)	//Actually put reagents in the mob
+				qdel(splash_holder)
+				var/mob/living/carbon/C = M
+				if(current_temperature <= 283.5) //Colder than 10 degrees is going to make you very cold
+					if(iscarbon(M))
+						C.adjust_bodytemperature(-80, 80)
+					to_chat(M, "<span class='warning'>The water is freezing cold!</span>")
+				else if(current_temperature >= 308.5) //Hotter than 35 celsius is going to make you burn up
```
Comment (utilbird) [inline code/modules/pool/pool.dm:266]: spas are ~40C, 35 shouldn't burn you, if realism is a concern xd

File: code/modules/pool/pool_items.dm [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+/obj/item/twohanded/required/pool/Initialize()
+	. = ..()
+	//Pick a random color
+	color = pick(COLOR_YELLOW, COLOR_LIME, COLOR_RED, COLOR_BLUE_LIGHT, COLOR_CYAN, COLOR_MAGENTA)
+
+/obj/item/twohanded/required/pool/check_float(mob/living/holder)
+	if(wielded)
+		return TRUE
+	return FALSE
```
Comment (utilbird) [inline code/modules/pool/pool_items.dm]:
```suggestion
	return wielded
```

File: code/modules/pool/pool.dm [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+	. += "<span class='boldnotice'>The thermostat on it reads [current_temperature].</span>"
+
+/obj/machinery/pool_filter/Initialize()
+	. = ..()
+	create_reagents(100, OPENCONTAINER) //If you're a terrible terrible clown and want to dump reagents into the pool.
+	if(preset_reagent_type)
+		reagents.add_reagent(preset_reagent_type, 100)
+	var/area/AR = get_area(src)
+	for(var/turf/open/indestructible/sound/pool/water in get_area_turfs(AR))
+		if(!istype(water))
```
Comment (utilbird) [inline code/modules/pool/pool.dm]: This check doesn't do much, The for() loop above will only ever grab water

**Comment by ike709** [comment]
🔑

File: code/modules/pool/pool.dm (hunk 8 above)
Comment (PowerfulBacon) [reply code/modules/pool/pool.dm:266]: spacemen are used to the coldness of space

**Review by utilbird — APPROVED** [review]

**Pull request merged**